### ISSUE TITLE
no-multi-spaces error thrown when lining up assignment default operators when destructuring

### ISSUE THREAD

**Issue by vhmth** [eslint/eslint#8015]
**Tell us about your environment**

* **ESLint Version:** 3.14.1
* **Node Version:** 4.7.0
* **npm Version:** 2.15.11

**What parser (default, Babel-ESLint, etc.) are you using?**

babel-eslint@7.1.1

**Please show your full configuration:**

```
{
  "extends": "eslint:recommended",
  "env": {
    "es6": true,
    "mocha": true,
    "node": true
  },
  "parser": "babel-eslint",
  "rules": {
    "arrow-spacing": "error",
    "block-spacing": "error",
    "curly": "error",
    "default-case": "error",
    "indent": [2, 2, {"SwitchCase": 1, "VariableDeclarator": {"var": 2, "let": 2, "const": 3}}],
    "newline-after-var": ["error", "always"],
    "no-console": ["error", {
      "allow": ["log", "warn", "error"]
    }],
    "no-debugger": "error",
    "no-else-return": "error",
    "no-extra-bind": "error",
    "no-implicit-coercion": "error",
    "no-multi-spaces": ["error", { "exceptions": { "VariableDeclarator": true, "AssignmentExpression": true } }],
    "no-template-curly-in-string": "error",
    "no-trailing-spaces": "warn",
    "no-undef-init": "error",
    "no-unused-vars": "error",
    "no-var": 1,
    "object-shorthand": "error",
    "prefer-const": "error",
    "prefer-reflect": "error",
    "semi": "error",
    "space-before-function-paren": ["error", {
      "anonymous": "always",
      "named": "never",
      "asyncArrow": "ignore"
    }],
    "spaced-comment": 1,
    "strict": "error"
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
const {
  dimensions = [],
  refresh    = true,
  unit       = UNIT_COUNT
} = options;
```

**What did you expect to happen?**

Per the `AssignmentExpression` exception being set to `true` in the my config, I figured that would extend to destructuring assignment expressions. I'm cool with a different rule (that may make sense), but my team would like to line up `=` defaults when destructuring objects.

**What actually happened? Please include the actual, raw output from ESLint.**

```
/Users/vinay/Code/loom/src/server/utilities/metrics.js
  87:16  error  Multiple spaces found before '='  no-multi-spaces
  88:16  error  Multiple spaces found before '='  no-multi-spaces

✖ 2 problems (2 errors, 0 warnings)
```

**Comment by mysticatea** [comment]
Thank you for the report.

This is correct behavior.
If you add `AssignmentPattern` as exceptions, probably you get the expected behavior.
`AssignmentPattern` is the AST node of default value's assignments in destructuring.

**Comment by vhmth** [comment]
Boom that works! :-) Thank you @mysticatea.
